### PR TITLE
Add alignment parameter to matrices and vectors

### LINKED ISSUE #454: Add alignment parameter to matrices and vectors
If I write the matrix

`$ mat(-1, 1, 1; 1, -1, 1; 1, 1, -1) $`

this is displayed as

image

To "correctly" display this, I currently have to add & to every single cell in the matrix, when all I really want is for cells to be aligned to be flush right

`$ mat(-1&, 1&, 1&; 1&, -1&, 1&; 1&, 1&, -1&) $`

image

While allowing arbitrary alignment is nice, it would be nicer in this case if I could write something like

`$ mat(-1, 1, 1; 1, -1, 1; 1, 1, -1; align: right) $`

to achieve the same effect. (Also for vectors)

In LaTeX I can do this with

`\[ \begin{pmatrix*}[r] -1 & 1 & 1\\ 1 & -1 & 1\\ 1 & 1 & -1 \end{pmatrix*} \]`

$$ \begin{pmatrix*}[r]
-1 & 1 & 1\\
1 & -1 & 1\\
1 & 1 & -1 \end{pmatrix*} $$

### PR DESCRIPTION
**Pull request by mkorje**
Closes #454. Adds an `align` parameter that takes a horizontal alignment to `math.mat` and `math.vec`. The way alignment points work in matrices and vectors hasn't been changed. However, if alignment points are used in a column (so a vector or a column in a matrix), the value of the `align` parameter will have no effect on that column's elements.

matrix

<details>
<summary>Typst code</summary>

```typst
#set page(width: 10cm, height: auto)

$ vec(-1, 1, -1, align: #left)
  vec(-1, 1, -1, align: #center)
  vec(-1, 1, -1, align: #right)
  vec(-&1, 1, -&1, align: #left)
  vec(-&1, 1, -&1, align: #center)
  vec(-&1, 1, -&1, align: #right) $

$ mat(-1, 1, 1; 1, -1, 1; 1, 1, -1; align: #left)
  mat(-1, 1, 1; 1, -1, 1; 1, 1, -1; align: #center)
  mat(-1, 1, 1; 1, -1, 1; 1, 1, -1; align: #right) $

$ mat(-1, 1, 1; 1, -&1, 1; 1, &1, -1; align: #left)
  mat(-1, 1, 1; 1, -&1, 1; 1, &1, -1; align: #center)
  mat(-1, 1, 1; 1, -&1, 1; 1, &1, -1; align: #right) $
```

</details>

There was some discussion on the [discord](https://discord.com/channels/1054443721975922748/1176478139757629563/1285427275516477573), where alignment points were shown to be useful in some scenarios. Perhaps in the future, if an improved design for matrices can be decided upon which absolves the need for alignment points, they can then be removed.

### CODE REVIEW

File: crates/typst/src/math/matrix.rs [hunk 1]
```diff
@@ -29,7 +30,8 @@ const DEFAULT_STROKE_THICKNESS: Em = Em::new(0.05);
 
 /// A column vector.
 ///
-/// Content in the vector's elements can be aligned with the `&` symbol.
+/// Content in the vector's elements can be aligned with the
+/// [`align`]($category/math/vec.align) parameter, or the `&` symbol.
```
Comment (laurmaedje) [inline crates/typst/src/math/matrix.rs:34]:
```suggestion
/// [`align`]($math.vec.align) parameter, or the `&` symbol.
```
_should_ work in principle. did you not try it or did it not work?

**Review by laurmaedje — COMMENTED** [review]
That's a surprisingly small change set!

File: crates/typst/src/math/matrix.rs (hunk 1 above)
Comment (mkorje) [reply crates/typst/src/math/matrix.rs:34]: Oh no I didn't try it. I just had no idea which worked and when I looked for some examples I could only find ones with `$category` prepended. I'll change it and try it after I finish with the stretch PR.

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**